### PR TITLE
Replace version.earlier and version.later with version.at_least

### PR DESCRIPTION
**Pull request by martica**
I'm not going to merge this until Dan is finished with #772. It'd be too disruptive. But the current system is too confusing for me.

## PR Checklist:

- [ ] All new features have been tested
- [ ] All new features have been documented
- [ ] I have read the **CONTRIBUTING.md** file
- [ ] I will abide by the code of conduct

### CODE REVIEW

**Comment by martica** [comment]
Dan's PR with 3.7 support is in. This is merged against that now and ready to go.

File: batavia/core/version.js [hunk 1]
```diff
@@ -36,11 +36,24 @@ version.version_id = function(str) {
...
+    if (constants.BATAVIA_MAGIC === undefined) {
+        throw new exceptions.BataviaError.$pyclass('Version test called before version defined.')
+    }
     return magic_map[constants.BATAVIA_MAGIC] < version.version_id(str)
 }
 
 version.later = function(str) {
+    if (constants.BATAVIA_MAGIC === undefined) {
+        throw new exceptions.BataviaError.$pyclass('Version test called before version defined.')
+    }
```
Comment (freakboy3742) [inline batavia/core/version.js]: Are earlier and later still being used anywhere? Is there any reason to keep them? Is there a need to have both "later" *and* "at_least"? (I agree at_least is a much easier to read name...)

Comment (martica) [reply batavia/core/version.js]: I was thinking about simultaneous in-flight PRs, but it looks pretty clean right now. I'll chuck them out now and simplify things.

Comment (martica) [reply batavia/core/version.js]: Hah. We introduced a bunch of new ones, obviously, in the 3.7 branch. Merging against that isn't sufficient to update those as well.

**Pull request merged**